test/JDBC/src/main/java/com/sqlsamples/JDBCBabelOutputConcurrency.java

```diff
@@ -0,0 +1,548 @@
+
+import org.apache.logging.log4j.Logger;
+import java.io.*;
+import java.util.*;
+import java.sql.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.sqlsamples.Config.*;
+import static com.sqlsamples.Statistics.curr_exec_time;
+
+public class JDBCBabelOutputConcurrency {
+    static AtomicInteger errorCount = new AtomicInteger(0);
+    
+    private static String initializeConnectionString() {
+        String url = properties.getProperty("URL");
+        String port = properties.getProperty("tsql_port");
+        String databaseName = properties.getProperty("databaseName");
+        String user = properties.getProperty("user");
+        String password = properties.getProperty("password");
+
+        return createSQLServerConnectionString(url, port, databaseName, user, password);
+    }
+
+    public static void runTest(BufferedWriter bw, Logger logger) {
+        long startTime = System.nanoTime();
+
+        try {
+            setupTestData(bw);
+            babel_output_concurrency_test(bw);
+            cleanupTestData(bw);
+            
+            if (errorCount.get() > 0) {
+                bw.write("Test failed with " + errorCount.get() + " errors");
+                bw.newLine();
+            }
+            else {
+                bw.write("Test passed");
+                bw.newLine();
+            }
+        } catch (Exception e) {
+            try {
+                bw.write(e.getMessage());
+                bw.newLine();
+            } catch (IOException ioe) {
+                ioe.printStackTrace();
+            }
+        }
+
+        long endTime = System.nanoTime();
+        curr_exec_time = endTime - startTime;
+    }
+    
+    private static void setupTestData(BufferedWriter bw) throws Exception {
+        String connectionString = initializeConnectionString();
+        try (Connection conn = DriverManager.getConnection(connectionString)) {
+            Statement stmt = conn.createStatement();
+            
+            // Create test table
+            stmt.execute("CREATE TABLE TestTable_jira_4880 (TestID INT, Param1 VARCHAR(50), LockProcessID INT DEFAULT 0)");
+            stmt.execute("INSERT INTO TestTable_jira_4880 VALUES (1, 'InitialValue', 100)");
+            stmt.execute("INSERT INTO TestTable_jira_4880 VALUES (2, NULL, 100)");
+            
+            bw.write("Test data created successfully");
+            bw.newLine();
+        }
+    }
+    
+    private static void cleanupTestData(BufferedWriter bw) throws Exception {
+        String connectionString = initializeConnectionString();
+        try (Connection conn = DriverManager.getConnection(connectionString)) {
+            Statement stmt = conn.createStatement();
+            stmt.execute("DROP TABLE IF EXISTS TestTable_jira_4880");
+        }
+    }
+    
+    /*
+     * JIRA-4880: This will be a concurrency test for OUTPUT clause operations to ensure no issues or crashes are encountered.
```

Comment:
Have you tried running this test without your changes and verified that we can reproduce a crash?